### ISSUE TITLE
freeze model for inference with output_node_name

### ISSUE THREAD

**Issue by AbhimanyuAryan** [tensorflow/tensorflow#30776]
I want to compile the TensorFlow Graph to Movidius Graph. I have used Model Zoo's `ssd_mobilenet_v1_coco` model to train it on my own dataset.

Then I ran 
```
python object_detection/export_inference_graph.py \
				--input_type=image_tensor \
				--pipeline_config_path=/home/redtwo/nsir/ssd_mobilenet_v1_coco.config \
                                --trained_checkpoint_prefix=/home/redtwo/nsir/train/model.ckpt-3362 \
				--output_directory=/home/redtwo/nsir/output
```
which generates me `frozen_interference_graph.pb` & `saved_model/saved_model.pb`

![11](https://user-images.githubusercontent.com/8083613/61330462-3fe5b180-a83d-11e9-99a5-7b63aa1b7d2a.png)

![12](https://user-images.githubusercontent.com/8083613/61330465-4116de80-a83d-11e9-89cb-2d69e213391b.png)

Now to convert this saved model into *Movidius graph*. There are commands given

Export GraphDef file
```
python3 ../tensorflow/tensorflow/python/tools/freeze_graph.py \
		--input_graph=inception_v3.pb \
		--input_binary=true \
		--input_checkpoint=inception_v3.ckpt \
		--output_graph=inception_v3_frozen.pb \
		--output_node_name=InceptionV3/Predictions/Reshape_1
```
Freeze model for inference
```
python3 ../tensorflow/tensorflow/python/tools/freeze_graph.py \
		--input_graph=inception_v3.pb \
		--input_binary=true \
		--input_checkpoint=inception_v3.ckpt \
		--output_graph=inception_v3_frozen.pb \
		--output_node_name=InceptionV3/Predictions/Reshape_1
```

which can finally be feed to **NCS Intel Movidius SDK**

```
mvNCCompile -s 12 inception_v3_frozen.pb -in=input -on=InceptionV3/Predictions/Reshape_1
```

All of this is given at Intel Movidius Website here: https://movidius.github.io/ncsdk/tf_modelzoo.html

#### Question: My model was already trained i.e. `output/frozen_inference_graph`. Why do I again freeze it using `/slim/export_inference_graph.py` or it's the `output/saved_model/saved_model.py` that will go as input to `slim/export_inference_graph.py`??

All I want is **output_node_name=Inceptionv3/Predictions/Reshape_1**. How to get this output_name_name directory structure & anything inside it? I don't know what all it contains 

------------------------

### System information
- **What is the top-level directory of the model you are using**: 
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: TensorFlow installed with pip
- **TensorFlow version (use command below)**: 1.13.1
- **Bazel version (if compiling from source)**: 
- **CUDA/cuDNN version**: V10.1.168/7.*
- **GPU model and memory**: 2080Ti 11Gb
- **Exact command to reproduce**:

**Comment by AbhimanyuAryan** [comment]
@rmothukuru @ymodak what exactly is this problem? Can you give me some instructions. I'm clueless & I have deadlines in place to get it to work. Thanks

**Comment by AbhimanyuAryan** [comment]
what output node should I use for model zoo's `ssd_mobilenet_v1_coco` model(trained on my own custom dataset)

```
python freeze_graph.py --input_graph=/path/to/graph.pbtxt --input_checkpoint=/path/to/model.ckpt-22480 --input_binary=false --output_graph=/path/to/frozen_graph.pb --output_node_names="the nodes that you want to output e.g. InceptionV3/Predictions/Reshape_1 for Inception V3 "
```

Things I understand & don't understand: 
input_checkpoint: ✓ [check points that were created during training]
output_graph: ✓ [path to output frozen graph]
out_node_names: X


I don't understand `out_node_names` parameter & what should inside this considering it's ssd_mobilnet not inception_v3

**Comment by AbhimanyuAryan** [comment]
'''
A typical freeze_graph.py command looks like this :

python freeze_graph.py --input_meta_graph model.ckpt.meta --output_node_names "Model/Activation_8/softmax_output" --output_graph model_frozen.pb --input_checkpoint model.ckpt --input_binary=true
'''

This answer is using a softmax at end. But I'm not sure what is my last layer/node 

https://software.intel.com/en-us/forums/computer-vision/topic/806596

**Comment by AbhimanyuAryan** [comment]
slim  has been discontinued by TF? https://github.com/tensorflow/models/issues/539#issuecomment-363954823 and NCS supports SLIM implementation.

**Comment by pumpkinband** [comment]
I want to make a “freeze_graph.pb” but I cannot understand graph and cannot find “output_node_names” can anyone please help me @AbhimanyuAryan @ymodak

**Comment by AbhimanyuAryan** [comment]
@pumpkinband it's yet not solved. Also I have switched to OpenVino. Since NCSDK v1 & v2 are both out of support

**Comment by ymodak** [comment]
You may also use [```netron```](https://github.com/lutzroeder/netron) for figuring out node names.
Tensorboard also provides the ability to visualize the network as you mentioned with [Op-level graph](https://www.tensorflow.org/tensorboard/graphs#op-level_graph)
